aya/src/bpf.rs

```diff
@@ -110,6 +110,17 @@ impl TryFrom<u32> for PinningType {
     }
 }
 
+impl TryFrom<bool> for PinningType {
```

Comment:
A `TryFrom` that never fails should be `From` (without the try part). But in this case I don't think we need a `From` either, I'd just create the right variant at the call site

Reply:
Agreed I originally just tried to use `PinningType` directly here but it was public only within it's crate, and I ended up with 

```
error[E0446]: crate-private type `bpf::PinningType` in public interface
    --> aya/src/obj/mod.rs:1308:1
     |
1308 | pub fn parse_map_info(info: bpf_map_info, pinned: PinningType) -> Map {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-private type
 ```
 
 Is the right move just to make this enum public?  (sorry I'm a bit of a rust noob :) )

Reply:
I don't think parse_map_info needs to be pub? Also as per comment, just move it to maps/mod.rs

Reply:
Well `BtfMapDef` is private to the `obj` crate which is why I put the helper function there

Reply:
And If I leave it in the `obj` crate is has to be public to be used from the `maps` crate right?

Reply:
`obj` is a module not a crate, so you can export symbols at the crate level with `pub(crate)`.

Since `BtfMapDef` is defined in `obj` I guess just leave `parse_map_info` in there too, and make it `pub(crate)`

Reply:
Ack that makes more sense, thanks for bearing with me!! :)